__tests__/utils/reference/resolveReferences.test.js

```diff
@@ -0,0 +1,223 @@
+/*
```

Comment:
Mostly copied from the resolveObject tests but I refactored every single test to make it a unit-test for the resolveReference function, which is now more granularly tested